kernel/src/engine/arrow_expression/mod.rs

```diff
@@ -310,8 +347,11 @@ impl ExpressionEvaluator for DefaultExpressionEvaluator {
             (expr, output_type) => {
                 let array_ref = evaluate_expression(expr, batch, Some(output_type))?;
                 let array_ref = apply_schema_to(&array_ref, output_type)?;
-                let arrow_type = ArrowDataType::try_from_kernel(output_type)?;
-                let schema = ArrowSchema::new(vec![ArrowField::new("output", arrow_type, true)]);
+                // Use the actual data type of the array, not the converted kernel type
```

Comment:
when does using `output_type` cause an issue? I feel like we should _not_ do this, because we'll get unexpected behavior where the output doesn't actually match what we asked for